Source/WebCore/dom/Document.cpp

```diff
@@ -9820,8 +9820,14 @@ CompositeOperator Document::compositeOperatorForBackgroundColor(const Color& col
 void Document::didAssociateFormControl(Element& element)
 {
     RefPtr page = this->page();
-    if (!page || !page->chrome().client().shouldNotifyOnFormChanges())
+    if (!page)
         return;
+    if (!page->chrome().client().shouldNotifyOnFormChanges()) {
+        // Is it sufficient to only support registering event listeners on the window object, or do we need to support other event targets?
```

Comment:
For autofill, and more generally for user scripts in safari, we add event listener on the window. So it's fine.